src/commands/synthetics/run-tests-lib.ts

```diff
@@ -191,53 +176,86 @@ const getTestListBySearchQuery = async (
     suite: `Query: ${testSearchQuery}`,
   }))
 }
-
-export const getTestsList = async (
+export const getTriggerConfigs = async (
   api: APIHelper,
   config: RunTestsCommandConfig,
   reporter: MainReporter,
-  suites: Suite[] = []
-) => {
-  // If "testSearchQuery" is provided, always default to running it.
-  if (config.testSearchQuery) {
-    const testsToTriggerBySearchQuery = await getTestListBySearchQuery(api, config.global, config.testSearchQuery)
-
-    if (testsToTriggerBySearchQuery.length > MAX_TESTS_TO_TRIGGER) {
-      reporter.error(
-        `More than ${MAX_TESTS_TO_TRIGGER} tests returned by search query, only the first ${MAX_TESTS_TO_TRIGGER} will be fetched.\n`
-      )
+  suites?: Suite[]
+): Promise<TriggerConfig[]> => {
+  const defaultTestConfigOverrides = config.global
+  // TODO: Clean up locations as part of SYNTH-12989
+  const testConfigOverridesFromEnv = config.locations?.length ? {locations: config.locations} : {}
+  const testsFromSearchQuery = config.testSearchQuery ? await getTestsFromSearchQuery(api, config, reporter) : []
+  const testsFromTestConfigs = await getTestConfigs(config, reporter, suites)
+
+  // Grab the list of publicIds of tests to trigger from config file/env variables/CLI params, search query or test config files
+  const testIdsFromCli = config.publicIds
+  const testIdsFromSearchQuery = testsFromSearchQuery.map(({id}) => id)
+  const testIdsFromTestConfigs = testsFromTestConfigs.map(({id}) => id)
+
+  const testIdsToTrigger =
+    testIdsFromCli.length !== 0
+      ? testIdsFromCli
+      : testIdsFromSearchQuery.length !== 0
+      ? testIdsFromSearchQuery
+      : testIdsFromTestConfigs.length !== 0
+      ? testIdsFromTestConfigs
+      : []
+
+  // Create the overrides required for the list of tests to trigger
+  const triggerConfigs = testIdsToTrigger.map((id) => {
+    const testFromSearchQuery = testsFromSearchQuery.find((test) => test.id === id)
+    const testFromTestConfigs = testsFromTestConfigs.find((test) => test.id === id)
+
+    return {
+      id,
+      ...testFromSearchQuery,
+      ...testFromTestConfigs,
+      config: {
+        ...defaultTestConfigOverrides,
+        ...testConfigOverridesFromEnv,
+        ...testFromTestConfigs?.config,
+      },
     }
+  })
 
-    return testsToTriggerBySearchQuery
+  return triggerConfigs
+}
+
+const getTestsFromSearchQuery = async (api: APIHelper, config: RunTestsCommandConfig, reporter: MainReporter) => {
+  const testsToTriggerBySearchQuery = await getTestListBySearchQuery(api, config.global, config.testSearchQuery || '')
+
+  if (testsToTriggerBySearchQuery.length > MAX_TESTS_TO_TRIGGER) {
+    reporter.error(
+      `More than ${MAX_TESTS_TO_TRIGGER} tests returned by search query, only the first ${MAX_TESTS_TO_TRIGGER} will be fetched.\n`
+    )
```

Comment:
Let's not change anything in this PR, but I wonder if this is necessary, since it's handled as well [here](https://github.com/DataDog/datadog-ci/blob/17e3c7f6e3a1726477db9b78c85f762010049314/src/commands/synthetics/utils/public.ts#L714).
Maybe we could simplify this later?